src/backend/legendary/library.ts

```diff
@@ -631,19 +632,23 @@ export class LegendaryLibrary {
   public hasGame = (appName: string) => this.allGames.has(appName)
 }
 
+const legendaryMutex = new Mutex()
+
 export async function runLegendaryCommand(
   commandParts: string[],
   abortController: AbortController,
   options?: CallRunnerOptions
 ): Promise<ExecResult> {
   const { dir, bin } = getLegendaryBin()
-  return callRunner(
-    commandParts,
-    { name: 'legendary', logPrefix: LogPrefix.Legendary, bin, dir },
-    abortController,
-    {
-      ...options,
-      verboseLogFile: legendaryLogFile
-    }
+  return legendaryMutex.runExclusive(async () =>
```

Comment:
I think instead of blocking everything, we could use a timeout here to avoid blocking other stuff.
Because I imagine that this is to avoid doing several requests to refresh the list of games due to re-render on some react component. But If we completely block the process, Heroic will be on an infinite loop when refreshing the library if a game is being downloaded if I understood this correctly.

Maybe implement this like on this example:
https://github.com/DirtyHairy/async-mutex#limiting-the-time-waiting-for-a-mutex-or-semaphore-to-become-available

Reply:
Problem is that we can't know if/when Legendary writes to its "sensitive" files.

As an example, every command that requires authentication might write to `user.json` at any point of its runtime.
Right now, I believe all of them authenticate at the start, so theoretically a timeout would work, but any change to that in Legendary would then break this system, potentially without us knowing before we release & some unlucky users with exactly the right system configuration are affected